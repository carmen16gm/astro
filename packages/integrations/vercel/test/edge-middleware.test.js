import { loadFixture } from './test-utils.js';
import { expect, use } from 'chai';
import chaiJestSnapshot from 'chai-jest-snapshot';

use(chaiJestSnapshot);

describe('Serverless prerender', () => {
	/** @type {import('./test-utils').Fixture} */
	let fixture;

	beforeEach(function () {
		chaiJestSnapshot.configureUsingMochaContext(this);
	});

	before(async () => {
		chaiJestSnapshot.resetSnapshotRegistry();
		fixture = await loadFixture({
			root: './fixtures/middleware/',
		});
	});

	it('build successfully the middleware edge file', async () => {
		await fixture.build();
		const contents = await fixture.readFile(
			// this is abysmal...
			'../.vercel/output/functions/render.func/packages/integrations/vercel/test/fixtures/middleware/dist/middleware.mjs'
		);
		expect(contents).to.matchSnapshot();
	});
});
