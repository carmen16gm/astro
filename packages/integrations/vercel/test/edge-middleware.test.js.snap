// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Serverless prerender build successfully the middleware edge file 1`] = `
"// test/fixtures/middleware/src/vercel-edge-middleware.js
function vercel_edge_middleware_default({ request, context }) {
  return {
    title: \\"Hello world\\"
  };
}

// test/fixtures/middleware/dist/middleware2.mjs
var onRequest = async (context, next) => {
  const response = await next();
  return response;
};

// <stdin>
import { createContext, trySerializeLocals } from \\"astro/middleware\\";
async function middleware(request, context) {
  const url = new URL(request.url);
  const ctx = createContext({
    request,
    params: {}
  });
  ctx.locals = vercel_edge_middleware_default({ request, context });
  const next = async () => {
    const response = await fetch(url, {
      headers: {
        \\"x-astro-locals\\": trySerializeLocals(ctx.locals)
      }
    });
    return response;
  };
  return onRequest(ctx, next);
}
export {
  middleware as default
};
"
`;
